providers/zohocrm/write.go

```diff
@@ -67,10 +69,13 @@ func (c *Connector) Write(ctx context.Context, config common.WriteParams) (*comm
 		write = c.Client.Post
 	}
 
-	// ZohoCRM requires everything to be wrapped in a "data" object.
-	// RecordData should be a list of map[string]any
-	body := map[string]any{
-		"data": config.RecordData,
+	// Checking if the config.RecordData asserts to []map[string]any
+	// If not, we wrap the data in a slice, else we send it as is.
+	recordDataCopy, ok := config.RecordData.([]map[string]any)
+	if !ok {
+		body = map[string]any{"data": []any{config.RecordData}}
```

Comment:
I would also assert `config.RecordData.(map[string]any)` here and error out if not ok.